pkg/config/config.go

```diff
@@ -695,6 +697,14 @@ type Destination struct {
 	IsMachine bool `json:",omitempty" toml:"is_machine,omitempty"`
 }
 
+// PodmanshConfig represents configuration for the podman shell
+type PodmanshConfig struct {
+	// Shell to start in container, default: "/bin/sh"
+	Shell string `toml:"shell"`
+	// Name of the container the podmansh user should join
+	Container string `toml:"container"`
```

Comment:
Add Timeout here and deprecate podmansh_timeout, and remove it from man page and containers.conf.

Reply:
I'm in the process of doing so and need input: Do we have somewhere an example value of a deprecated value? Otherwise I would keep the old value in the internal struct and use this one, if the new value is zero.

Reply:
I will use https://github.com/containers/common/blob/477496cfee44c5f9a4bc69c59c56a47ccb84fe9d/pkg/config/config.go#L341 as a template and keep the old value, while making it deprecated.

Reply:
Done. @rhatdan @Luap99 please have a look at my suggestion.

I had to fiddle with the backwards compatability of `podmansh_timeout`. I wanted to have a solution, where `podmansh.Timeout` is preferred over the now deprecated `podmansh_timeout` value and the best idea I had was to use a default 0 value for both variables when they are not set. Then the `newLocked` function can use either of the two, if they are set and otherwise apply the default setting of 30.

This solution works, but it appears a bit clumsy. If any of you have a better idea for a more elegant solution please share.

Will work on the man page once we have resolved the remaining issues.